### PR TITLE
Rename MaxScheduleTime to ScheduleTimeoutSeconds

### PR DESCRIPTION
**Pull request by cwdsuzhou**
/cc @Huang-Wei

### CODE REVIEW

File: kep/42-podgroup-coscheduling/README.md [hunk 1]
```diff
@@ -50,8 +50,8 @@ type PodGroupSpec struct {
 	// will not start anyone.
 	MinResources *v1.ResourceList `json:"minResources,omitempty"`
 
-	// MaxScheduleTime defines the maximal time of members/tasks to wait before run the pod group;
-	MaxScheduleTime *metav1.Duration `json:"maxScheduleTime,omitempty"`
+	// MaxScheduleTimeSeconds defines the maximal time of members/tasks to wait before run the pod group;
+	MaxScheduleTime *metav1.Duration `json:"maxScheduleTimeSeconds,omitempty"`
```
Comment (Huang-Wei) [inline kep/42-podgroup-coscheduling/README.md:54]: According to [api-conventions#units](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#units), the type should be an integer. Let's make it `*int32`.

And based on:

> We use the fooSeconds convention for durations, as discussed in the units subsection.
> - fooPeriodSeconds is preferred for periodic intervals and other waiting periods (e.g., over fooIntervalSeconds).
> - fooTimeoutSeconds is preferred for inactivity/unresponsiveness deadlines.
> - fooDeadlineSeconds is preferred for activity completion deadlines.

How about renaming it to `ScheduleTimeoutSeconds`?

Comment (cwdsuzhou) [reply kep/42-podgroup-coscheduling/README.md:54]: Seem better. Updated

PTAL

File: pkg/util/podgroup.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ import (
 	"sigs.k8s.io/scheduler-plugins/pkg/apis/podgroup/v1alpha1"
 )
 
-// DefaultWaitTime is 60s if MaxScheduleTime is not specified.
+// DefaultWaitTime is 60s if ScheduleTimeoutSeconds is not specified.
```
Comment (Huang-Wei) [inline pkg/util/podgroup.go:30]: Forgot to mention that a good practice is to use `Defaults_XYZ` logic to define the expected defaulting behavior, and register the the function to `RegisterDefaults()` (in pkg/apis/podgroup/v1alpha1/zz_generated.defaults.go).

Feel free to do it in this PR, or add a TODO here and accomplish in a follow-up.

**Review by Huang-Wei — COMMENTED** [review]
Apologizing for verbose comments here: would you mind updating `uint32` to `int32`? According to [api-conventions.md#primitive-types](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types):

> Do not use unsigned integers, due to inconsistent support across languages and libraries. Just validate that the integer is non-negative if that's the case.

And for `MinMember`, what if users leave it unset, can the PodGroup be created successfully then?

File: pkg/util/podgroup.go (hunk 2 above)
Comment (cwdsuzhou) [reply pkg/util/podgroup.go:30]: I would like  add a TODO here.

**Comment by cwdsuzhou** [comment]
> Apologizing for verbose comments here: would you mind updating `uint32` to `int32`? According to [api-conventions.md#primitive-types](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types):
> 
> > Do not use unsigned integers, due to inconsistent support across languages and libraries. Just validate that the integer is non-negative if that's the case.
> 
> And for `MinMember`, what if users leave it unset, can the PodGroup be created successfully then?

I think we can add default value and validation in crd.yaml.
In this PR, I have changed uint32 to int32

File: pkg/util/podgroup.go (hunk 2 above)
Comment (cwdsuzhou) [reply pkg/util/podgroup.go:30]: PR updated

**Comment by Huang-Wei** [comment]
Thanks @cwdsuzhou !

/lgtm
/approve

**Pull request merged**